### PR TITLE
CRM-21212 Include getUfId function for wordpress to help getting E2E …

### PR DESCRIPTION
**Pull request by seamuslee001**
…unit tests running for wordpress

----------------------------------------
* CRM-21212: Get E2E Tests working on wordpress
  https://issues.civicrm.org/jira/browse/CRM-21212

ping @totten

### CODE REVIEW

**Comment by seamuslee001** [comment]
also ping @kcristiano are you able to validate if this looks right to you?

**Comment by seamuslee001** [comment]
@totten I think this will leave 1 E2E test that doesn't seem to be working for me locally but maybe environmental issues

File: CRM/Utils/System/WordPress.php [hunk 1]
```diff
@@ -355,6 +355,20 @@ public function permissionDenied() {
   }
 
   /**
+   * Determine the native ID of the CMS user.
+   *
+   * @param string $username
+   * @return int|NULL
+   */
+  public function getUfId($username) {
+    $userdata = get_user_by('login', $username);
```
Comment (kcristiano) [inline CRM/Utils/System/WordPress.php]: @seamuslee001 is there a reason you want to use get_user_by() as opposed to get_current_user_id() ?

Comment (seamuslee001) [reply CRM/Utils/System/WordPress.php]: @kcristiano I saw its what is used in the function just above in L334, The purpose of this i think is that in the E2E tests we are passing through a user name + pw and authenticating then wanting to get the id back

Comment (kcristiano) [reply CRM/Utils/System/WordPress.php]: You are correct - I was not thinking about the proper context.  As we are running tests we need to pas in the user name = pw.

**Comment by seamuslee001** [comment]
@totten I think given Kevin's comments i think this is good to merge

**Comment by totten** [comment]
Thank you, @seamuslee001 @kcristiano. :)

For anyone reading this later, it may not be very clear what `getUfId()` is about. APIv3 has a feature where you can provide a username in lieu of a contact ID. Compare:

 * `cv api contact.get id=123`
 * `cv api contact.get id=@user:demo`

The feature relies on `getUfId()` -- which wasn't implemented on WordPress. #11017 adds `getUfId()`, which fixes the support for `@user:<name>`, which in turn makes the E2E suite more compatible with WordPress.

**Pull request merged**